README.md

```diff
@@ -24,5 +24,169 @@ Note that the RTI website has [Free Trial](https://www.rti.com/free-trial) offer
+```
+
+If the environment variable is set, if a profile name matches the fully qualified node name (e.g.: `/my/full/namespace/my_node_name`), it will be used instead of the default profile.
+For example:
+```xml
+<?xml version="1.0"?>
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+xsi:noNamespaceSchemaLocation="http://community.rti.com/schema/5.3.1/rti_dds_qos_profiles.xsd" version="5.3.1">
+  <qos_library name="Ros2TestQosLibrary">
+    <qos_profile name="Ros2TestDefaultQos" base_name="BuiltinQosLib::Baseline.5.3.0" is_default_qos="true">
+      <participant_qos>
+        <property>
+          <value>
+            <!-- 6.25 MB/sec (52 Mb/sec) flow controller -->
+            <element>
+              <name>dds.flow_controller.token_bucket.slow_flow.token_bucket.max_tokens</name>
+              <value>8</value>
+            </element>
+            <element>
+              <name>dds.flow_controller.token_bucket.slow_flow.token_bucket.tokens_added_per_period</name>
+              <value>8</value>
+            </element>
+            <element>
+              <name>dds.flow_controller.token_bucket.slow_flow.token_bucket.bytes_per_token</name>
+              <value>8192</value>
+            </element>
+            <element>
+              <name>dds.flow_controller.token_bucket.slow_flow.token_bucket.period.sec</name>
+              <value>0</value>
+            </element>
+            <element>
+              <name>dds.flow_controller.token_bucket.slow_flow.token_bucket.period.nanosec</name>
+              <value>10000000</value>
+            </element>
+          </value>
+        </property>
+      </participant_qos>
+    </qos_profile>
+    <qos_profile name="/my_node" base_name="BuiltinQosLib::Baseline.5.3.0">
+      <datawriter_qos topic_filter="rt/my_large_data_topic">
+        <publish_mode>
+          <flow_controller_name>dds.flow_controller.token_bucket.slow_flow</flow_controller_name>
+        </publish_mode>
+      </datawriter_qos>
+    </qos_profile>
+  </qos_library>
+</dds>
+```
+
+Will use the `slow_flow` flow controller for publishers in the `my_large_data_topic` of the `/my_node` node.
+All other nodes will use the default profile.
+
+The profiles matching the node name will only be used for `datawriter` and `datareader` qos policies.
+Participant/Publisher/Subscription policies will always use the default profile.
+
+RTI Connext will log an error each time that it tries to find a profile that doesn't exist, if you didn't add a profile for each node those logs will appear.
+
+If you only provided one QoS library to the process, that one will be used.
+If not, the `RMW_CONNEXT_QOS_PROFILE_LIBRARY` must be used:
+```bat
+:: Windows
+set RMW_CONNEXT_QOS_PROFILE_LIBRARY=Ros2TestQosLibrary
+```
+```bash
+# Linux/MacOS
+export RMW_CONNEXT_QOS_PROFILE_LIBRARY=Ros2TestQosLibrary
+```
+
+### Ignoring the ROS specified QoS
+
+If you want to override the QoS profile specified in ROS code, you can do that by setting:
```

Comment:
Is setting an environment variable required in order to override a QoS? From my understanding of the text above, the ROS QoS will be used unless I have a profile with an appropriate `topic_filter` attribute, in which case that topic will use the specified QoS without any "opt-in" environment variable being set. 
If `RMW_CONNEXT_IGNORE_ROS_QOS` is set, then all topics will use the default QoS, unless there is a QoS with an appropriate `topic_filter` attribute. 
Is that correct?

Reply:
> Is setting an environment variable required in order to override a QoS?

It's not needed for setting things like "Flow controllers", which ROS doesn't know about.
But it's needed to override the things included in the ROS QoS profile.

> From my understanding of the text above, the ROS QoS will be used unless I have a profile with an appropriate topic_filter attribute, in which case that topic will use the specified QoS without any "opt-in" environment variable being set.

To be more clear:
```xml
    <qos_profile name="default" is_default_qos="true">
      <datawriter_qos topic_filter="rt/my_topic">
        <reliability>reliable</reliability>
      </datawriter_qos>
    </qos_profile>
```
would not override the reliability QoS for publishers in topic "/my_topic" except:
- ros reliability is set to "system default"
- RMW_CONNEXT_IGNORE_ROS_QOS is set.

> If RMW_CONNEXT_IGNORE_ROS_QOS is set, then all topics will use the default QoS, unless there is a QoS with an appropriate topic_filter attribute.

Yes (and there could be also a profile matching the name of the node).
This might be a bit error prone, because the ROS specified qos profiles are never used.

---

This is maybe a bit different of what I originally said in the meeting.
I diverged a bit because of wanting to use the "topic filters" feature, but maybe that was not a good idea.

The reason why `RMW_CONNEXT_IGNORE_ROS_QOS` is need is because there's no way to check if you're using the default qos or one that matches a topic filter (Connext doesn't provide an API to query that).

If profile names are used instead of the topic filter feature, then it's possible to do the following:
- Use the profile matching the topic name (if there's one).
- Use the ros profile, if there's no profile matching the name.

In that case it's not possible to use different QoS in different nodes for the same topic, but that maybe isn't really useful.
This alternative may be less error prone to the current PR.

There's another issue: though you can query if a profile of a given name exists, if it doesn't exist Connext will always logs an error (which will generate some noise in the logs at startup).
The only way to avoid the logs is to temporally deactivate Connext logging, that might be bad because you can lose an error log from another thread (the logger verbosity is not "thread local").

---

To sum up:
- We can use the approach of this PR, and set `RMW_CONNEXT_IGNORE_ROS_QOS`.
  That requires to carefully specify "all" QoS policies in the file (at least, all the ones that cannot be the default).
- Use the approach here, and set all ROS QoS policies that should be overridden to "system default".
  Because of "topic filters", you can choose different profiles for different topics.
- Don't use topic filters and look for a profile name matching the topic name. That will not allow wildcards, but it won't require setting the ros profile to "system default" in the places where an override is desired and will be less error prone that using the `RMW_CONNEXT_IGNORE_ROS_QOS` environment variable.

@mallanmba which alternative sounds more desirable?
Using profile names instead of topic filters is an easy change.

cc @jacobperron

Reply:
Updated after yesterday discussion in https://github.com/ros2/rmw_connext/pull/460/commits/1aae68ac33cdd31a5882d16e32753bc87171c844.

The current approach allows you to specify QoS in two ways:
- A profile marked `is_default_true="true"` with topic filters. In that case you cannot override non-system default QoS policies, but specifying things like a flow controller is possible. Doesn't require any environment variable set.
- If a profile matching the DDS topic name (see name mangling) is found, those will override the ROS QoS.
  Requires environment variable `RMW_CONNEXT_ALLOW_TOPIC_QOS_PROFILES=1`.